### DISCUSSION TITLE
Flux with resource definitions in PVC

### DISCUSSION THREAD

**Discussion by tobias-jenkner** [fluxcd/flux2#3493] [General] [answered]
Is it somehow possible to have flux access resource definitions stored in a PVC (without a git repository) or alternatively access a git repository stored in a PVC ?

**Comment by makkes** [comment, marked as answer]
Consuming resources from a PVC would violate the GitOps principle of resources being "[Versioned and Immutable](https://opengitops.dev/)".

If you want to keep resources in the cluster, you can e.g. spin up an S3-compatible storage solution like MinIO and put the resources into a [bucket](https://fluxcd.io/flux/components/source/buckets/) there.
